### PR TITLE
bug fix - html rendering when multiple percentiles are the same 

### PR DESCRIPTION
**Pull request by sephib**
## Summary

Fixes a bug when an HTML benchmark reports is generated. The HTML report crashes in the browsers with  e.g. `Error: xs[2] == xs[3]` when rendering distributions with consecutive duplicate percentile values.

This occurs when benchmarks have limited data points (e.g., < 10 requests), causing multiple percentiles to collapse to the same value. The visualization library expects strictly increasing x-values and fails when encountering duplicates.

### Root Cause

When a benchmark has very few successful requests (e.g., 3 requests), percentile calculations result in many duplicate values:

```python
# Before fix - causes visualization error
{
  "p001": 15.288,
  "p01": 15.288,   # duplicate
  "p05": 15.288,   # duplicate
  "p10": 15.288,   # duplicate
  "p25": 15.288,   # duplicate
  "p50": 16.413,   # unique
  "p75": 16.413,   # duplicate
  "p90": 17.035,   # unique
  ...
}

# After fix - renders successfully
{
  "p001": 15.288,
  "p50": 16.413,
  "p90": 17.035
}
```

## Details

- Added `_filter_duplicate_percentiles()` helper function to remove consecutive duplicate percentile values
- Override `model_dump()` in `_TabularDistributionSummary` to automatically apply filtering during HTML generation
- Added comprehensive unit tests (8 test cases) covering all edge cases

### Unit Tests
```bash
uv run pytest tests/unit/benchmark/test_html_output.py -v
# 8/8 tests passing
```

---

- [ X] "I certify that all code in this PR is my own, except as noted below."

## Use of AI

- [ x] Includes AI-assisted code completion
- [ x] Includes code generated by an AI application
- [ x] Includes AI-generated tests (NOTE: AI written tests should have a docstring that includes `## WRITTEN BY AI ##`)

### CODE REVIEW

File: src/guidellm/benchmark/outputs/html.py [hunk 1]
```diff
@@ -222,6 +240,35 @@ def _create_html_report(js_data: dict[str, str], output_path: Path) -> Path:
...
+
+    filtered = {}
+    previous_value = None
+
+    for key in percentile_order:
+        if key in percentiles:
+            current_value = percentiles[key]
+            if previous_value is None or current_value != previous_value:
+                filtered[key] = current_value
+                previous_value = current_value
```
Comment (sjmonson) [inline src/guidellm/benchmark/outputs/html.py]: Mathematically its more accurate to keep the largest percentile. Can you switch to keeping the largest?

Comment (sephib) [reply src/guidellm/benchmark/outputs/html.py]: updated to keep the largest

**Comment by sephib** [comment]
Hi @sjmonson 
1. i've rebased the branch
2. run  pre-commit - passes
3. there are some files that do not pass (but not related to this PR)
```
        modified:   src/guidellm/data/builders.py
        modified:   src/guidellm/data/config.py
        modified:   src/guidellm/data/entrypoints.py
        modified:   src/guidellm/data/schemas.py
        modified:   tests/unit/data/deserializers/test_synthetic.py
        modified:   tests/unit/data/test_builders.py
 ```
       
Please let me know if there any other issues to attend

**Comment by sjmonson** [comment]
Srry, the commit history was broken due to a mix of rebase and merge commits and I accidentally closed the PR while fixing it. Will re-open in a new PR.
